public/components/common/tag_filter_popover_content/tag_filter_popover_content.tsx

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useMemo, useCallback } from 'react';
+import {
+  EuiPopoverTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiToken,
+  EuiText,
+  EuiSpacer,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiFieldNumber,
+} from '@elastic/eui';
+import { TagValueSelector } from './tag_value_selector';
+
+interface TagKey {
+  name: string;
+  type: 'string' | 'number';
+}
+
+export enum TagFilterOperator {
+  Is = 'is',
+  IsNot = 'is not',
+  IsGreaterThan = 'is greater than',
+  IsLessThan = 'is less than',
+  IsOneOf = 'is one of',
+  IsNotOneOf = 'is not one of',
+}
+
+export interface TagFilterValue {
+  name: string;
+  operator: TagFilterOperator;
+  value: string | string[] | number;
+}
+
+export interface TagFilterPopoverContentProps {
+  tagFilter?: TagFilterValue;
+  tagKeys: TagKey[];
+  onCancel: () => void;
+  onSave: (tagFilter: TagFilterValue) => void;
+}
+
+const getValueInput = (
+  operator: TagFilterOperator,
+  valueType: 'string' | 'number',
+  value: string | string[] | number | undefined,
+  onChange: (value: string | string[] | number | undefined) => void
+) => {
+  if (valueType === 'string') {
+    return (
+      <TagValueSelector
+        singleSelection={operator === TagFilterOperator.Is || operator === TagFilterOperator.IsNot}
+        value={typeof value === 'number' ? undefined : value}
+        onChange={onChange}
+      />
+    );
+  }
+  return (
+    <EuiFieldNumber
+      value={value === undefined ? '' : value}
+      onChange={(e) => {
+        const newValue = e.currentTarget.value;
+        onChange(newValue === '' ? undefined : parseFloat(newValue));
+      }}
+      fullWidth
+      placeholder="Add a value"
+    />
+  );
+};
+
+export const TagFilterPopoverContent = ({
+  onSave,
+  tagKeys,
+  onCancel,
+  tagFilter,
+}: TagFilterPopoverContentProps) => {
+  const [value, setValue] = useState<string | string[] | number | undefined>(tagFilter?.value);
+  const [selectedTagOptions, setSelectedTagOptions] = useState<
+    Array<EuiComboBoxOptionOption<TagKey>>
+  >(() => {
+    if (!tagFilter) {
+      return [];
+    }
+    return [
+      {
+        label: tagFilter.name,
+        value: {
+          name: tagFilter.name,
+          type: typeof tagFilter.value === 'number' ? 'number' : 'string',
```

Comment:
The tag should have its own `type`, maybe should that instead of infer from the the `value`?

Reply:
The `type` is unused in the filter API level, when we apply search condition in the model group search API. We won't use the `type` field. I just want to reduce the passed fields. We can add it back.